lib/interactive_viewer.ml

```diff
@@ -134,3 +141,14 @@ let view (patches : Patch.t list) =
     ]
 
 let start patch = Ui_loop.run ~quit ~tick_period:0.2 (view patch)
+let convert_char_to_key (c : char) : Ui.key = (`Uchar (Uchar.of_char c), [])
+
+let start_test patch events =
+  let content_ui = Lwd.quick_sample (Lwd.observe (view patch)) in
```

Comment:
When you call `quick_sample`, you need to call `quick_release` when you want the variables to be updated again.

Reply:
Additionally, you need to call `quick_sample` after every events to allow the changes to propagate in the UI.
See the function `run_with_term` in `vendor/lwd/lib/nottui/nottui.ml` line 837 that implements an event loop. The function `step` calls `Lwd.quick_sample` and `Renderer.update` and handles events.
You need to call `quick_release` only once, when you're done observing the root.